comments_blackbox_test.go

```diff
@@ -240,3 +240,27 @@ func (s *CommentsSuite) TestUpdateCommentWithOtherUser() {
 	userSvc, _, _, commentsCtrl := s.securedControllers(testsupport.TestIdentity2)
 	test.UpdateCommentsForbidden(s.T(), userSvc.Context, userSvc, commentsCtrl, commentId, updateCommentPayload)
 }
+
+func (s *CommentsSuite) TestDeleteCommentWithSameAuthenticatedUser() {
+	// given
+	workitemId := s.createWorkItem(testsupport.TestIdentity)
+	commentId := s.createWorkItemComment(testsupport.TestIdentity, workitemId)
+	userSvc, _, _, commentsCtrl := s.securedControllers(testsupport.TestIdentity)
+	test.DeleteCommentsOK(s.T(), userSvc.Context, userSvc, commentsCtrl, commentId)
+}
+
+func (s *CommentsSuite) TestDeleteCommentWithoutAuth() {
+	// given
+	workitemId := s.createWorkItem(testsupport.TestIdentity)
+	commentId := s.createWorkItemComment(testsupport.TestIdentity, workitemId)
+	userSvc, commentsCtrl := s.unsecuredController()
+	test.DeleteCommentsUnauthorized(s.T(), userSvc.Context, userSvc, commentsCtrl, commentId)
+}
+
+func (s *CommentsSuite) TestDeleteCommentWithOtherAuthenticatedUser() {
+	// given
+	workitemId := s.createWorkItem(testsupport.TestIdentity)
+	commentId := s.createWorkItemComment(testsupport.TestIdentity, workitemId)
+	userSvc, _, _, commentsCtrl := s.securedControllers(testsupport.TestIdentity2)
+	test.DeleteCommentsUnauthorized(s.T(), userSvc.Context, userSvc, commentsCtrl, commentId)
```

Comment:
Actually, this should be a 'Forbidden' error (403) because the user is not allowed to perform the operation. "Unauthorized" error applies when the user is not authenticated (the "unauthorized" term can be confusing, though).